star_wars_magic_methods/people.py

```diff
@@ -1,3 +1,61 @@
 class People(object):
     def __init__(self, name, dark_side=False):
-        pass
+        self.name = name
+        self.dark_side = dark_side
+        self.light_side = not self.dark_side
+
+    def __str__(self):
+        return self.name
+        
+    def __neg__(self):
+        self.dark_side = True
+        self.light_side = False
+        
+    def __pos__(self):
+        self.dark_side = False
+        self.light_side = True
+        
+    def __invert__(self):
+        self.dark_side = not self.dark_side
+        self.light_side = not self.light_side
+        
+    def __xor__(self, other):
+        return "{} force chokes {}.".format(self.name, other.name)
+        
+    __rxor__ = __xor__
+        
+    def __eq__(self, other):
+        
+        if other.name == 'Han Solo':
+            return "{} shoots {}. BECAUSE HAN SHOOTS FIRST.".format(other.name, self.name)
+        
+        return "{} shoots {}.".format(self.name, other.name)
+        
+    def __call__(self):
+        return "Help me {}, you're my only hope.".format(self.name)
+        
+    def __truediv__(self, other):
+        if type(other) is not People:
+            raise TypeError()
+            
+        return '{} swings a lightsaber at {}.'.format(self.name, other.name)
+    
+    __div__ = __truediv__
```

Comment:
I like this line

Reply:
thanks @asungo , i steal the idea from the mentors